Module 2. CRUD/src/services/users.service.ts

```diff
@@ -0,0 +1,71 @@
+import { User } from "../models/user.interface";
+import { v4 as uuidv4 } from 'uuid';
+
+export class UserService {
+    private users: Array<User> = [
+        {
+            id: uuidv4(),
+            login: 'user1',
+            password: 'password1',
+            age: 20,
+            isDeleted: false
+        },
+        {
+            id: uuidv4(),
+            login: 'user2',
+            password: 'password2',
+            age: 20,
+            isDeleted: false
+        },
+        {
+            id: uuidv4(),
+            login: 'user3',
+            password: 'password3',
+            age: 20,
+            isDeleted: false
+        },
+    ];
+
+    getUsers(): Promise<Array<User>> {
+        return Promise.resolve(this.users.filter(user => !user.isDeleted));
+    }
+
+    getUserById(id: string): Promise<User | undefined> {
+       return Promise.resolve(this.users.find(user => user.id === id));
+    }
+
+    createUser(user: User): Promise<User | string> {
+        if(this.users.some(x => x.login === user.login)) {
+           return Promise.reject(`User with ${user.login} login already exists`);
+        }
+        this.users.push({...user, id: uuidv4()});
+        return Promise.resolve(user);
+    }
+
+    updateUser(user: User): Promise<User | string> {
+        if(this.users.some(x => x.login === user.login)) {
+            return Promise.reject(`User with ${user.login} login already exists`);
+        }
+        const index = this.users.findIndex(x => x.id === user.id);
+        this.users[index] = user;
+        return Promise.resolve(user);
+    }
+
+    deleteUser(id: string): Promise<User | string> {
+        const user = this.users.find(x => x.id === id);
+        if(user && !user.isDeleted) {
+            user.isDeleted = true;
+            return Promise.resolve(user);
+        } else {
+            return Promise.reject('User does not exists');
+        }
+    }
+
+    getAutoSuggestUsers(login: string, limit: number): Promise<Array<User>> {
+        return Promise.resolve(
+            this.users.filter((user: User) => user.login.includes(login))
```

Comment:
Please do early return when limit is set. Google for how to do this or ask me in work channel